Updated readme with useful info (#23)
submariner-io/cloud-prepare: README.md

# Submariner cloud-prepare

<!-- markdownlint-disable line-length -->
[![Linting](https://github.com/submariner-io/cloud-prepare/workflows/Linting/badge.svg)](https://github.com/submariner-io/cloud-prepare/actions?query=workflow%3ALinting)
<!-- markdownlint-enable line-length -->

Submariner's cloud-prepare is a Go library that provides API and capabilities for setting up cloud infrastructure in order to install
Submariner.

## API

The [main API](https://github.com/submariner-io/cloud-prepare/blob/devel/pkg/api/api.go) defines the capabilities for any `Cloud`:

* Preparing the cloud for setting up Submariner.
* Cleaning up the cloud after Submariner has been uninstalled.

These capabilities aim to be idempotent, so in case of failure or other necessity they are safe to re-run.

The API defines a `Reporter` type which has the capability to report on the latest operation performed in the cloud.

### Prepare a cloud for Submariner

> Preparing a cloud for Submariner takes in number of gateways, internal ports for intra-cluster communications (Submariner components)
> and public ports for inter-cluster communications (Submariner to Submariner).

Reviewer: ```suggestion
The `PrepareForSubmarinerInput` function takes the number of gateways, the internal ports used for intra-cluster communication between Submariner components,
and the public ports used for inter-cluster communication between Submariner gateways.
```

```go
	input := api.PrepareForSubmarinerInput{
		InternalPorts: []api.PortSpec{
			{Port: vxlanPort, Protocol: "udp"},
			{Port: metricsPort, Protocol: "tcp"},
		},
		PublicPorts: []api.PortSpec{
			{Port: nattPort, Protocol: "udp"},
			{Port: natDiscoveryPort, Protocol: "udp"},
		},
		Gateways: gateways,
	}
	err := cloud.PrepareForSubmariner(input, reporter)

```

### Clean up a cloud after Submariner has been uninstalled

> Cleanup doesn't take input as it finds all preparation work done by the library and reverses it.

Reviewer: ```suggestion
The `CleanupAfterSubmariner` function reverses all the preparation work previously done by the library.
```

```go

> 	err := cloud.CleanupAfterSubmariner(repoter)

Reviewer: ```suggestion
	err := cloud.CleanupAfterSubmariner(reporter)
```

```

> 
> ## AWS
> 
> AWS is supported by the library. In order to prepare an AWS instance, it needs to have OpenShift pre-installed and running.

Reviewer: ```suggestion

## Supported Cloud Providers

### AWS

In order to prepare an AWS instance, it needs to have OpenShift pre-installed and running.

```
